presto-hive/src/main/java/io/prestosql/plugin/hive/s3select/S3SelectPushdown.java

```diff
@@ -85,7 +87,20 @@ private static boolean isSerdeSupported(Properties schema)
     private static boolean isInputFormatSupported(Properties schema)
     {
         String inputFormat = getInputFormatName(schema);
-        return SUPPORTED_INPUT_FORMATS.contains(inputFormat);
+
+        if (TextInputFormat.class.getName().equals(inputFormat)) {
+            if (!Objects.equals(schema.getProperty(SKIP_HEADER_COUNT_KEY, "0"), "0")) {
+                // S3 Select supports skipping one line of headers, but it was returning incorrect results for presto-hive-hadoop2/conf/files/test_table_with_header.csv.gz
```

Comment:
Maybe shorten to "for gzip files"
```java
// S3 Select supports skipping one line of headers, but it was returning incorrect results for gzip files
```
Or just the file name `test_table_with_header.csv.gz`

Reply:
I want to refer to an exact file. I checked that it works correctly some some .gz files, so it looks data-dependent.
I will leave this as-is, and will add a TODO + link to the issue.

Reply:
> I will leave this as-is,

(primarily because i missed this comment earlier)